cmd/csi-external-health-monitor-agent/main.go

```diff
@@ -0,0 +1,209 @@
+)
+
+const (
+
+	// Default timeout of short CSI calls like GetPluginInfo
+	csiTimeout = time.Second
+)
+
+// Command line flags
+var (
+	monitorInterval = flag.Duration("monitor-interval", 1*time.Minute, "Interval for controller to check volumes health condition.")
+
+	kubeconfig      = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	resync          = flag.Duration("resync", 10*time.Minute, "Resync interval of the controller.")
+	csiAddress      = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	showVersion     = flag.Bool("version", false, "Show version.")
+	timeout         = flag.Duration("timeout", 15*time.Second, "Timeout for waiting for attaching or detaching the volume.")
+	workerThreads   = flag.Uint("worker-threads", 10, "Number of pv monitor worker threads")
+	kubeletRootPath = flag.String("kubelet-root-path", "/var/lib/kubelet", "The root path pf kubelet.")
+
+	metricsAddress = flag.String("metrics-address", "", "The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled.")
+	metricsPath    = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+)
+
+var (
+	version = "unknown"
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	if *showVersion {
+		fmt.Println(os.Args[0], version)
+		return
+	}
+	klog.Infof("Version: %s", version)
+
+	// Create the client config. Use kubeconfig if given, otherwise assume in-cluster.
+	config, err := buildConfig(*kubeconfig)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	if *workerThreads == 0 {
+		klog.Error("option -worker-threads must be greater than zero")
+		os.Exit(1)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	factory := informers.NewSharedInformerFactory(clientset, *resync)
+
+	metricsManager := metrics.NewCSIMetricsManager("" /* driverName */)
+
+	// Connect to CSI.
+	csiConn, err := connection.Connect(*csiAddress, metricsManager, connection.OnConnectionLoss(connection.ExitOnConnectionLoss()))
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	err = rpc.ProbeForever(csiConn, *timeout)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	// Find driver name.
+	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
+	defer cancel()
+	storageDriver, err := rpc.GetDriverName(ctx, csiConn)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	klog.V(2).Infof("CSI driver name: %q", storageDriver)
+	metricsManager.SetDriverName(storageDriver)
+	metricsManager.StartMetricsEndpoint(*metricsAddress, *metricsPath)
+
+	supportGetNodeVolumeHealth, err := supportGetNodeVolumeHealth(ctx, csiConn)
```

Comment:
It's renamed to VOLUME_CONDITION in CSI spec, so change to supportNodeGetVolumeCondition